go-controller/pkg/networkmanager/nad_controller.go

```diff
@@ -413,11 +413,23 @@ func (c *nadController) GetActiveNetworkForNamespace(namespace string) (util.Net
 	if !util.IsNetworkSegmentationSupportEnabled() {
 		return &util.DefaultNetInfo{}, nil
 	}
+
+	// check if required UDN label is on namespace
+	ns, err := c.namespaceLister.Get(namespace)
```

Comment:
nit: I think this snippet is coming here a second time after I saw it in cluster-manager, time for a new util called "ValidUDNNamespace" or something which get's the namespace and checks for the label, might be useful for the future

Reply:
I think it's ok for now. The logic is pretty short and the decision made on the logic is different in each case.